tracing/src/test/java/com/palantir/tracing/TracersTest.java

```diff
@@ -257,6 +264,120 @@ public void testWrapRunnable_startsNewSpan() throws Exception {
         assertThat(Tracer.completeSpan().get().getOperation()).isEqualTo("outside");
     }
 
+    @Test
+    public void testWrapFuture_immediate() {
+        List<Span> observed = new ArrayList<>();
+        Tracer.subscribe("futureTest", observed::add);
```

Comment:
shall we just make this a TestSubscriber? it would be a teeny class but might make all of our unit-tests a little shorter!